### DISCUSSION TITLE
Turkish Q Keyboard Layout and Clipboard Paste

### DISCUSSION THREAD

**Discussion by akcansoft** [86Box/86Box#5012] [Show and tell]
https://github.com/akcansoft/86Box-TR-Keyboard
Turkish Q Keyboard Support and Clipboard Paste Functionality in the 86box Emulator

This AutoHotkey v2 script is designed to enable the Turkish Q keyboard layout in the 86box emulator.
It maps keys from the US keyboard layout to their corresponding characters in the Turkish Q layout, making it possible to use the Turkish keyboard seamlessly.

Additionally, the script supports automatically pasting clipboard content into the emulator using the Ctrl+V combination.
To ensure correct character transfer, Unicode text from the clipboard is converted to Turkish DOS (IBM 857) encoding, and each character is sent to the emulator.
This approach provides a faster and more accurate typing and data-pasting experience.

**Comment by OBattler** [comment]
You should instead configure the guest OS to use the turkish keyboard layout. In DOS, that would be KEYB TR.

**Reply by akcansoft** [reply]
There is no Turkish keyboard support in versions prior to DOS 5. Also, some other systems do not have Turkish keyboard support.
